### PR TITLE
Jwt integration

### PR DESCRIPTION
**Pull request by robertavram**
TODO: views need help to not filter out records for the Unicef user

### CODE REVIEW

File: django_api/django_api/settings/base.py [hunk 1]
```diff
@@ -295,3 +340,38 @@
...
+# JWT Authentication
+# production overrides for django-rest-framework-jwt
+if not DISABLE_JWT_AUTH:
+    public_key_text = open(os.path.join(BASE_DIR, 'keys/jwt/certificate.pem'), 'rb').read()  # noqa: F405
+    certificate = load_pem_x509_certificate(public_key_text, default_backend())
+
+    JWT_PUBLIC_KEY = certificate.public_key()
+
+    JWT_AUTH.update({  # noqa: F405
+        'JWT_SECRET_KEY': SECRET_KEY,
```
Comment (chachra) [inline django_api/django_api/settings/base.py]: not defined?

File: django_api/django_api/apps/unicef/permissions.py [hunk 2]
```diff
@@ -27,3 +27,21 @@ def has_permission(self, request, view):
...
+
+class UnicefPartnershipManagerOrRead(BasePermission):
+    """
+    Partner authorized officer and editor are allowed to change PD calculation
+    method in PRP.
+    """
+
+    def has_permission(self, request, view):
+        user = request.user
+        if user.is_authenticated() and request.method in SAFE_METHODS:
```
Comment (chachra) [inline django_api/django_api/apps/unicef/permissions.py]: Wouldn't this short-circuit the logic below?

File: django_api/django_api/apps/unicef/permissions.py [hunk 3]
```diff
@@ -27,3 +27,21 @@ def has_permission(self, request, view):
...
+            return True
+
+        return user.is_authenticated() and \
+            hasattr(user, 'jwt_payload') and \
+            "groups" in user.jwt_payload and \
+            "Partnership Manager" in user.jwt_payload['groups'] or \
```
Comment (chachra) [inline django_api/django_api/apps/unicef/permissions.py]: ("Partnership Manager" in user.jwt_payload['groups'] or \
 +            user.is_superuser)

to keep it readable / less error-prone...

File: django_api/django_api/settings/base.py (hunk 1 above)
Comment (robertavram) [reply django_api/django_api/settings/base.py]: defined on line 29.. it's the secret key used to sign the sessions in django

File: django_api/django_api/apps/unicef/permissions.py (hunk 2 above)
Comment (robertavram) [reply django_api/django_api/apps/unicef/permissions.py]: no... this is only for safe methods... if post.. then circuit below is important

File: django_api/django_api/apps/unicef/permissions.py (hunk 3 above)
Comment (robertavram) [reply django_api/django_api/apps/unicef/permissions.py]: I wanted user.is_superuser to be on par with everything else.. I'll add the parenthesis for the condition before the "or"

**Pull request merged**